From 3d04d5b678c2f95975d1639dc21398b79308af54 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alexander=20Hansen=20F=C3=A6r=C3=B8y?= <ahf@torproject.org>
Date: Wed, 28 Jan 2026 17:55:02 +0100
Subject: [PATCH 1/2] Explicitly define `SIO_UDP_NETRESET` for MinGW builds.

This patch explicitly defines the value `SIO_UDP_NETRESET` according to
both what Windows and ReactOS does.

Fixes: #29818.
---
 ssl/quic/quic_reactor.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/ssl/quic/quic_reactor.c b/ssl/quic/quic_reactor.c
index a754f285bbe2b..d8ac969d02a69 100644
--- a/ssl/quic/quic_reactor.c
+++ b/ssl/quic/quic_reactor.c
@@ -76,6 +76,12 @@ void ossl_quic_reactor_cleanup(QUIC_REACTOR *rtor)
 }
 
 #if defined(OPENSSL_SYS_WINDOWS)
+
+/* Work around for MinGW builds. */
+#if defined(__MINGW32__) && !defined(SIO_UDP_NETRESET)
+#  define SIO_UDP_NETRESET _WSAIOW(IOC_VENDOR, 15)
+#endif
+
 /*
  * On Windows recvfrom() may return WSAECONNRESET when destination port
  * used in preceding call to sendto() is no longer reachable. The reset

From 3136648b96665ca3fb1e622dd8f120f814e004c9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alexander=20Hansen=20F=C3=A6r=C3=B8y?= <ahf@torproject.org>
Date: Thu, 29 Jan 2026 16:36:49 +0100
Subject: [PATCH 2/2] fixup! Explicitly define `SIO_UDP_NETRESET` for MinGW
 builds.

---
 ssl/quic/quic_reactor.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ssl/quic/quic_reactor.c b/ssl/quic/quic_reactor.c
index d8ac969d02a69..deec428152c58 100644
--- a/ssl/quic/quic_reactor.c
+++ b/ssl/quic/quic_reactor.c
@@ -79,7 +79,7 @@ void ossl_quic_reactor_cleanup(QUIC_REACTOR *rtor)
 
 /* Work around for MinGW builds. */
 #if defined(__MINGW32__) && !defined(SIO_UDP_NETRESET)
-#  define SIO_UDP_NETRESET _WSAIOW(IOC_VENDOR, 15)
+#define SIO_UDP_NETRESET _WSAIOW(IOC_VENDOR, 15)
 #endif
 
 /*
